From: Stanislaw Gruszka <sgruszka@redhat.com>
Date: Fri, 3 Sep 2010 12:23:39 -0400
Subject: [wireless] fixes from 2.6.32.13
Message-id: <1283516623-11659-7-git-send-email-sgruszka@redhat.com>
Patchwork-id: 28070
O-Subject: [RHEL5 PATCH 06/10] wireless fixes from 2.6.32.13
Bugzilla: 621105
RH-Acked-by: David S. Miller <davem@redhat.com>
RH-Acked-by: Thomas Graf <tgraf@redhat.com>

Included commits:
fb85b86ad4989b8d3aa5cbb2aafc9568a22fd3ce mac80211: remove bogus TX agg state assignment

Omitted commits:
d5d30e59fbea20fc09d300f3d4c90bbd2ecfc9f3 p54usb: Add usbid for Corega CG-WLUSB2GT.
9121999db5232e1d81270df5d0787df1492408e7 p54pci: fix bugs in p54p_check_tx_ring
63ab8b7b6c1b21af040a2c8321092bce11bb561f p54pci: rx frame length check
8b0744c9cb38e6beb06ec17fc78e372e3b05e067 drivers/net/wireless/p54/txrx.c Fix off by one error

diff --git a/net/mac80211/agg-tx.c b/net/mac80211/agg-tx.c
index 843b5d7..5fed5a6 100644
--- a/net/mac80211/agg-tx.c
+++ b/net/mac80211/agg-tx.c
@@ -181,7 +181,6 @@ static void sta_addba_resp_timer_expired(unsigned long data)
 		       HT_AGG_STATE_REQ_STOP_BA_MSK)) !=
 						HT_ADDBA_REQUESTED_MSK) {
 		spin_unlock_bh(&sta->lock);
-		*state = HT_AGG_STATE_IDLE;
 #ifdef CONFIG_MAC80211_HT_DEBUG
 		printk(KERN_DEBUG "timer expired on tid %d but we are not "
 				"(or no longer) expecting addBA response there",
